lib/javascript/JavascriptParser.js

```diff
@@ -1799,6 +1805,7 @@ class JavascriptParser extends Parser {
 		for (let index = 0, len = statements.length; index < len; index++) {
 			const statement = statements[index];
 			this.walkStatement(statement);
+			if (this.scope.terminated) break;
```

Comment:
I like this solution, I'm only afraid of one thing, that some plugins could rely on it and rewrite something, and now we just will not handle them at all

Reply:
but this code is unreachable.. so I dont see a reason to waste cpu =)

Reply:
Yeah, I agree with you, I'm just worried that someone could use similar logic, but we can merge and look at feedback, It's not difficult to do a revert